sdk/rust/simple-searcher/src/main.rs

```diff
@@ -0,0 +1,220 @@
+    pub api_key: Option<String>,
+}
+
+
+async fn random() -> U256 {
+    let mut rng = rand::thread_rng();
+    U256::from(rng.gen::<u128>())
+}
+
+async fn submit_opportunity(
+    ws_client: WsClient,
+    opportunity: Opportunity,
+    private_key: String,
+) -> Result<()> {
+    let bid = match opportunity {
+        opportunity::Opportunity::Evm(opportunity) => {
+            Client::new_bid(
+                opportunity,
+                BidParamsEvm {
+                    amount:   U256::from(5_000_000_000_000_000_000_i128),
+                    nonce:    random().await,
+                    deadline: U256::from(
+                        (OffsetDateTime::now_utc() + Duration::days(1)).unix_timestamp(),
+                    ),
+                },
+                private_key,
+            )
+            .await
+        }
+        opportunity::Opportunity::Svm(opportunity) => {
+            Client::new_bid(opportunity, 2, private_key).await
+        }
+    }
+    .map_err(|e| {
+        eprintln!("Failed to create bid: {:?}", e);
+        anyhow!("Failed to create bid: {:?}", e)
+    })?;
+
+    let result = ws_client.submit_bid(bid).await;
+    match result {
+        Ok(_) => println!("Bid submitted"),
+        Err(e) => eprintln!("Failed to submit bid: {:?}", e),
+    };
+    Ok(())
+}
+
+async fn handle_opportunity(ws_client: WsClient, opportunity: Opportunity, args: RunOptions) {
+    // Assess the opportunity to see if it is worth bidding
+    // For the sake of this example, we will always bid
+    let private_key = match opportunity {
+        Opportunity::Evm(_) => {
+            println!("EVM opportunity Received");
+            args.private_key_evm.clone()
+        }
+        Opportunity::Svm(_) => {
+            println!("SVM opportunity Received");
+            args.private_key_svm.clone()
+        }
+    };
+
+    match private_key {
+        Some(private_key) => {
+            if let Err(e) = submit_opportunity(ws_client, opportunity, private_key).await {
+                eprintln!("Failed to submit opportunity: {:?}", e);
+            }
+        }
+        None => {
+            eprintln!("Private key not provided");
+        }
+    }
+}
+
+#[tokio::main]
+async fn main() -> Result<()> {
+    let args: RunOptions = RunOptions::parse();
+    if let Some(private_key) = args.private_key_evm.clone() {
+        private_key
+            .parse::<LocalWallet>()
+            .map_err(|e| anyhow!("Invalid evm private key: {}", e))?;
+    }
+
+    let client = Client::try_new(ClientConfig {
+        http_url: args.server_url.clone(),
+        api_key:  args.api_key.clone(),
+    })
+    .map_err(|e| {
+        eprintln!("Failed to create client: {:?}", e);
+        anyhow!("Failed to create client")
+    })?;
+
+    let ws_client = client.connect_websocket().await.map_err(|e| {
+        eprintln!("Failed to connect websocket: {:?}", e);
+        anyhow!("Failed to connect websocket")
+    })?;
+
+    let opportunities = client
```

Comment:
maybe move the next few lines into a separate function like "bid_on_existing_opps" or ...